### ISSUE TITLE
bfile.rsids() variant order is inconsistent with variant order in the bgen file

### ISSUE THREAD

**Issue by abarysh** [jeremymcrae/bgen#4]
Hello,

(Thank you for a great package.)

I noticed that `bfile.rsids()` and `bfile.positions()` return values in alphabetical order by rsid. This order is different from the order of variants in the bgen file itself (as shown by iteration). If this is intended behavior, it would be great to add a note to the documentation.

Here's an example:

```
>> bfile = BgenReader(bgen_path, delay_parsing=True)
>> rsids = bfile.rsids()
>> print(rsids) # prints variants in alphabetical order

['rs201106462', 
'rs367896724', 
'rs531646671', 
'rs531730856', 
'rs540538026', 
'rs541940975', 
'rs544419019', 
'rs546169444', 
'rs575272151', 
'rs62028691', 
'rs62635286']

>> for var in bfile:
>>     print(var.rsid) # prints variants in order of storage in the bgen file

rs367896724
rs201106462
rs575272151
rs544419019
rs540538026
rs62635286
rs62028691
rs531730856
rs546169444
rs531646671
rs541940975
```

Thank you,
A.

**Comment by jeremymcrae** [comment]
I think this depends on whether the bgen file has an associated index file (*.bgi). If an index file exists, bfile.rsids() returns rsIDs according to their order in the index file, which can differ arbitrarily from the variant order in the bgen file. Without an index file bfile.rsids() and variant iteration order should match.

I'll put a note in the readme, thanks!

And you can verify this via the example files in the test folder of this repo:
```py
from bgen import BgenReader

path_with_index = '.../bgen/tests/data/example.16bits.bgen'
path_without_index = '.../bgen/tests/data/example.15bits.bgen'

bfile_with_index = BgenReader(path_with_index)
bfile_without_index = BgenReader(path_without_index)

rsids_with_index = bfile_with_index.rsids()
rsids_without_index = bfile_without_index.rsids()

# rsIDs from the indexed file correspond to the order in the index, whereas rsIDs from 
# the unindexed file correspond to variant order, which match the iteration order
assert rsids_with_index != rsids_without_index
assert sorted(rsids_with_index) == sorted(rsids_without_index)

# iterating across both files shows same rsID order
for x, y in zip(bfile_with_index, bfile_without_index):
    print(x.rsid, y.rsid)
```

**Comment by abarysh** [comment]
Great, thank you! Is it possible that BgenWriter sorts variants before generating the bgi file? Because my test case is the following. I used BgenWriter to generate a custom bgen file (a subset of a larger source file) with an accompanying bgi file (generated automatically). When I use BgenReader to read my custom bgen file back, the variant order in the bgen file and bgi file are different.

**Comment by jeremymcrae** [comment]
Sure, when reading a bgen file with an associated index, it will now sort the rsids(), positions() and chroms() methods by the order they appear in the bgen file. This didn't end up changing the bgi file (which is a sqlite3 database). I just changed the relevant sql queries to order by the file offset of each variant.

I've pushed a new version which should work (v1.5.9), install that and let me know if there are any problems remaining?
